### PR TITLE
feat: I am in redirect hell aka frontend auth

### PR DESCRIPTION
**Pull request by AbhiPrasad**
This is not good code but it works

Ima abstract it - but opening to give yall a sense of my approach

### CODE REVIEW

**Comment by AbhiPrasad** [comment]
The irony when I didn't run `yarn fix` on my own PR

File: ui/src/pages/Login.tsx [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import Button from "@material-ui/core/Button";
 import Card from "@material-ui/core/Card";
 import CardContent from "@material-ui/core/CardContent";
+import CircularProgress from "@material-ui/core/CircularProgress";
```
Comment (hanlinc27) [inline ui/src/pages/Login.tsx:4]: Can we make it `import { Button, Card, CardContent, CircularProgress} from "@material-ui/core`

File: ui/src/pages/Login.tsx [hunk 2]
```diff
@@ -32,7 +32,29 @@ interface LoginProps {
...
+        setLoading(false);
+      });
+  }, [setLoading, history]);
+
+  if (loading) {
+    return <CircularProgress />;
+  }
+
+  function handleOnClick() {
+    window.location.href = `${window.location.origin}/api/login`;
```
Comment (hanlinc27) [inline ui/src/pages/Login.tsx:56]: for this function, when you have an invalid email and it directs to the error screen, clicking the button again doesn't seem to launch the sign-in flow for me?
image

File: ui/src/App.tsx [hunk 3]
```diff
@@ -2,16 +2,22 @@ import * as React from "react";
...
 
 const defaultFetcher = (
   input: RequestInfo,
   init?: RequestInit
 ): Promise<unknown> =>
   fetch(input, init)
-    .then((res) => res.json())
-    .then(({ payload }) => payload);
+    .then((res) => {
+      console.log(res);
```
Comment (hanlinc27) [inline ui/src/App.tsx:13]:
```suggestion
```

**Review by hanlinc27 — COMMENTED** [review]
Works fine for me w/ Blueprint email, the behaviour for invalid email button click isn't working. 
It won't go to `${window.location.origin}/api/login`; and button click continues to land on `/unauthorized`

File: restapi/api.go [hunk 4]
```diff
@@ -65,25 +65,25 @@ func Router(db *gorm.DB, locationFinder location.LocationFinder) (http.Handler,
...
 	r.Group(func(r chi.Router) {
 		r.Use(jwtauth.Verifier(config.GetJWTKey()))
 		r.Use(Authenticator)
 
-		// rest.PostHandler(r, "/stories", api.CreateStories)
-		// rest.PostHandler(r, "/story", api.CreateStoriesFormData)
-		// rest.PostHandler(r, "/authors", api.CreateAuthors)
+		rest.GetHandler(r, "/check_auth", api.checkAuth)
+		rest.PostHandler(r, "/stories", api.CreateStories)
+		rest.PostHandler(r, "/story", api.CreateStoriesFormData)
```
Comment (Shehryar21) [inline restapi/api.go:77]: With this change now and these endpoints being protected, how would i test these endpoints through Postman? What would I need to do to be authorized?

Comment (AbhiPrasad) [reply restapi/api.go:77]: You would need to hit `/api/login`, login with your blueprint email, and then grab the `jwt` and `oauthstate` cookies -> 
image

File: ui/src/pages/Login.tsx (hunk 2 above)
Comment (AbhiPrasad) [reply ui/src/pages/Login.tsx:56]: Huh weird, seems to work fine with me.

File: ui/src/pages/Login.tsx (hunk 1 above)
Comment (AbhiPrasad) [reply ui/src/pages/Login.tsx:4]: I always prefer to explicitly import for component libraries to make sure tree shaking works properly.

File: ui/src/pages/Login.tsx (hunk 2 above)
Comment (hanlinc27) [reply ui/src/pages/Login.tsx:56]: Tried on another browser and doesn't seem to work?

https://user-images.githubusercontent.com/19617248/111402922-73c44a80-86a2-11eb-882d-c6d758a3545e.mov

Comment (AbhiPrasad) [reply ui/src/pages/Login.tsx:56]: Ah yeah I see, lemme fix

**Pull request closed without merging**